src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -85,6 +85,7 @@ private AuthorityInfo(string host, string canonicalAuthority, AuthorityType auth
         public AuthorityType AuthorityType { get; }
         public string UserRealmUriPrefix { get; }
         public bool ValidateAuthority { get; }
+        public bool AutoDetectRegion { get; set; }
```

Comment:
I'm wondering a setter can be avoided here to keep this object immutable.

Reply:
This Object is created at app level and the AutoDetectRegion is set at request level. And hence the setter is required. I need the value in request context to retrieve it in Instance discovery manager. Could not figure out a better way.

Reply:
Ok, please note that AuthorityInfo has a copy constructor that needs to be updated : https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/241666c6756e30ca58bd88ed1e43c58203654f62/src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs#L73

Reply:
Added to the constructor.